discovery-provider/src/tasks/index_spl_token.py

```diff
@@ -170,53 +174,87 @@ def parse_spl_token_transaction(
 
-        # Balance is expected to change if there is a transfer instruction.
-        if receiver_token_account not in balance_changes:
-            logger.error(
-                f"index_spl_token.py | {tx_sig} error while parsing pre and post balances"
+        if version == 0:
+            balance_changes = get_solana_tx_token_balance_changes(
+                meta=meta, account_keys=account_keys, mint=WAUDIO_MINT
             )
-            return None
-        if balance_changes[receiver_token_account]["change"] == 0:
-            logger.error(f"index_spl_token.py | {tx_sig} no balance change found")
-            return None
-
-        receiver_spl_tx_info: SplTokenTransactionInfo = {
-            "user_bank": receiver_token_account,
-            "signature": tx_sig,
-            "slot": result.slot,
-            "timestamp": datetime.datetime.utcfromtimestamp(
-                float(result.block_time or 0)
-            ),
-            "vendor": vendor,
-            "prebalance": balance_changes[receiver_token_account]["pre_balance"],
-            "postbalance": balance_changes[receiver_token_account]["post_balance"],
-            "sender_wallet": sender_root_account,
-            "root_accounts": [sender_root_account, receiver_root_account],
-            "token_accounts": [sender_token_account, receiver_token_account],
-        }
-        return receiver_spl_tx_info
+            tx_infos: list[SplTokenTransactionInfo] = []
+            sender_token_accounts: list[str] = []
+            sender_root_accounts: list[str] = []
+            for key, balance_change in balance_changes.items():
+                if balance_change["change"] < 0:  # Only get senders
+                    sender_token_accounts.append(key)
+                    sender_root_accounts.append(balance_changes[key]["owner"])
+
+            for key, balance_change in balance_changes.items():
+                if balance_change["change"] > 0:  # Only get receivers
+                    receiver_spl_tx_info = {
+                        "user_bank": key,
+                        "signature": tx_sig,
+                        "slot": result.slot,
+                        "timestamp": datetime.datetime.utcfromtimestamp(
+                            float(result.block_time or 0)
+                        ),
+                        "vendor": vendor,
+                        "prebalance": balance_changes[key]["pre_balance"],
+                        "postbalance": balance_changes[key]["post_balance"],
+                        "root_accounts": sender_root_accounts
+                        + [balance_changes[key]["owner"]],
+                        "token_accounts": sender_token_accounts + [key],
+                        # The specific sender may be ambigous in the case of a Address Lookup Table
+                        "sender_wallet": None,
+                    }
+                    tx_infos.append(receiver_spl_tx_info)
+            return tx_infos
```

Comment:
seems like with a v0 tx we're now indexing both sender and receiver under spl_token, just making sure that's correct? i think in the old world sender falls under user bank indexer?

Reply:
nope! we are only indexing receiver here, we are just collecting the senders in case we have to enqueue them in the user balance refresh!